crates/fs/private/file.rs

```diff
@@ -1,24 +1,31 @@
 use chrono::{DateTime, Utc};
-use serde::{Deserialize, Serialize};
+use semver::Version;
+use serde::{de::Error as DeError, ser::Error as SerError, Deserialize, Deserializer, Serialize};
 
-use crate::{HashOutput, Id, Metadata, UnixFsNodeKind};
+use crate::{dagcbor, HashOutput, Id, Metadata, NodeType, UnixFsNodeKind};
 
-use super::{namefilter::Namefilter, INumber, PrivateNodeHeader};
+use super::{namefilter::Namefilter, INumber, Key, PrivateNodeHeader, RatchetKey, Rng};
 
 //--------------------------------------------------------------------------------------------------
 // Type Definitions
 //--------------------------------------------------------------------------------------------------
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct PrivateFileContent {
-    pub(crate) metadata: Metadata,
-    pub(crate) content: Vec<u8>, // Inlined file content. // TODO(appcypher): Support linked file content.
+#[derive(Debug, Clone, PartialEq)]
+pub struct PrivateFile {
+    pub r#type: NodeType,
```

Comment:
Same idea here, we can implicitly know `r#type` from the type, so we can remove the field.